odilo.py

```diff
@@ -0,0 +1,581 @@
+# coding=utf-8
+import base64
+import datetime
+import isbnlib
+import json
+import logging
+
+from sqlalchemy.orm.session import Session
+
+from httplib import HTTPException
+
+from model import (
+    get_one_or_create,
+    Collection,
+    Contributor,
+    Credential,
+    DataSource,
+    DeliveryMechanism,
+    Edition,
+    ExternalIntegration,
+    Hyperlink,
+    Identifier,
+    Representation,
+    Subject,
+)
+
+from core.util.http import RequestNetworkException
+
+from core.analytics import Analytics
+
+from metadata_layer import (
+    CirculationData,
+    ContributorData,
+    FormatData,
+    IdentifierData,
+    Metadata,
+    LinkData,
+    ReplacementPolicy,
+    SubjectData,
+)
+
+from coverage import (
+    BibliographicCoverageProvider,
+)
+
+from config import (
+    CannotLoadConfiguration,
+)
+
+from testing import DatabaseTest
+
+from util.http import (
+    HTTP,
+    BadResponseException,
+)
+
+
+class OdiloAPI(object):
+    log = logging.getLogger("Odilo API")
+
+    LIBRARY_API_BASE_URL = u"library_api_base_url"
+    BASE_URL = "http://localhost:8080/api/v2"  # Debug value by default
+
+    # --- OAuth ---
+    TOKEN_ENDPOINT = BASE_URL + "/token"
+
+    # --- Discovery API ---
+    ALL_PRODUCTS_ENDPOINT = BASE_URL + "/records"
+
+    RECORD_METADATA_ENDPOINT = BASE_URL + "/records/{recordId}"
+    RECORD_AVAILABILITY_ENDPOINT = BASE_URL + "/records/{recordId}/availability"
+
+    # --- Circulation API ---
+    CHECKOUT_ENDPOINT = BASE_URL + "/records/{recordId}/checkout"
+    CHECKOUT_GET = BASE_URL + "/checkouts/{checkoutId}"
+    CHECKIN_ENDPOINT = BASE_URL + "/checkouts/{checkoutId}/return"
+    # Downloads given checkout offering the url where to consume the digital resource.
```

Comment:
FYI we call this 'fulfillment' to make it clear that it's a separate operation from 'checkout'.

Reply:
yes, we know the difference and have also the two steps: checkout a resource, and the fulfill it in order to consume it. Maybe we should rename the API endpoint in the future to avoid confusion.